core/src/test/scala/unit/kafka/utils/JsonTest.scala

```diff
@@ -62,5 +65,26 @@ class JsonTest {
     assertEquals("{\"a\":1,\"b\":2}", Json.encode(Map("a" -> 1, "b" -> 2)))
     assertEquals("{\"a\":[1,2],\"c\":[3,4]}", Json.encode(Map("a" -> Seq(1,2), "c" -> Seq(3,4))))
   }
-  
+
+  @Test
+  def testParseTo() = {
+    val foo = "baz"
+    val bar = 1
+
+    val result = Json.parseTo(s"""{"foo": "$foo", "bar": $bar}""", classOf[TestObject])
+
+    assertTrue(result.isRight)
+    assertEquals(TestObject(foo, bar), result.right.get)
+  }
+
+  @Test
+  def testParseTo_invalidJson() = {
+    val result = Json.parseTo("{invalid json}", classOf[TestObject])
+
+    assertTrue(result.isLeft)
+    assertEquals(classOf[JsonParseException], result.left.get.getClass)
+  }
+
 }
+
+case class TestObject(@JsonProperty("foo") foo: String, @JsonProperty("bar") bar: Int)
```

Comment:
It's better to place this inside a companion object so that it's restricted to this test.